src/main/java/com/nirima/jenkins/plugins/docker/DockerTemplateBase.java

```diff
@@ -567,7 +567,13 @@ public void setExtraDockerLabelsString(String extraDockerLabelsString) {
 
     public DockerRegistryEndpoint getRegistry() {
         if (registry == null) {
-            registry = new DockerRegistryEndpoint(null, pullCredentialsId);
+            /* TODO: Use DockerRegistryEndpoint#fromImageName when JENKINS-63243 is fixed */
+            String hostname = NameParser.resolveRepositoryName(
+                NameParser.parseRepositoryTag(getImage()).repos).hostname;
+            if(!hostname.startsWith("https://")) {
+                hostname = "https://" + hostname;
```

Comment:
Are you 100% sure that `https://` is the correct default here?
I know that docker registries _can_ use https, and if one is authenticating then it's best to use an encrypted means of communicating, but it looks like the code here would prevent someone from using `"http://hostname.domain"` as this would be turned into `"https://http://hostname.domain"` which is unlikely to work at all.

I _suspect_ that it'd be better to only prefix with `https://` if there was no protocol specified at all ... but I'm not entirely sure of that here (I'm not in a position to experiment/evaluate right now either - current work pressures are such that my time for this is limited to "review, ask questions, push merge button and hope").

Reply:
> I suspect that it'd be better to only prefix with https:// if there was no protocol specified at all ...

[NameParser#resolveRepositoryName](https://github.com/docker-java/docker-java/blob/3.1.5/src/main/java/com/github/dockerjava/core/NameParser.java#L103) actually throw an exception if you provide the scheme in the image name. That is to say that `http://hostname.domain/repo/name:tag` is not a valid image name. So you cannot specify `http://`.
However [NameParser#resolveRepositoryName](https://github.com/docker-java/docker-java/blob/3.1.5/src/main/java/com/github/dockerjava/core/NameParser.java#L109) use `https://index.docker.io/v1/` if there no registry specify, hence the condition.
I agree that we could just do this if there is no protocol at all.

> Are you 100% sure that https:// is the correct default here?

[DockerRegistryEndpoint](https://github.com/jenkinsci/docker-commons-plugin/blob/docker-commons-1.17/src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java#L128) is also setting `https://` for the scheme. The help says `The url is built from registry:port into https://registry:port, the same way docker push does.`.